qiskit/algorithms/evolvers/variational/var_qte.py

```diff
@@ -0,0 +1,235 @@
+            imag_part_tol: Allowed value of an imaginary part that can be neglected if no
+                imaginary part is expected.
+            num_instability_tol: The amount of negative value that is allowed to be
+                rounded up to 0 for quantities that are expected to be
+                non-negative.
+            quantum_instance: Backend used to evaluate the quantum circuit outputs.
+        """
+        super().__init__()
+        self.variational_principle = variational_principle
+        self._quantum_instance = None
+        if quantum_instance is not None:
+            self.quantum_instance = quantum_instance
+        self.expectation = expectation
+        self.ode_function_factory = ode_function_factory
+        self.ode_solver_callable = ode_solver_callable
+        self.lse_solver_callable = lse_solver_callable
+        self.imag_part_tol = imag_part_tol
+        self.num_instability_tol = num_instability_tol
+
+    @property
+    def quantum_instance(self) -> Optional[QuantumInstance]:
+        """Returns quantum instance."""
+        return self._quantum_instance
+
+    @quantum_instance.setter
+    def quantum_instance(self, quantum_instance: Union[QuantumInstance, Backend]) -> None:
+        """Sets quantum_instance"""
+        if not isinstance(quantum_instance, QuantumInstance):
+            quantum_instance = QuantumInstance(quantum_instance)
+
+        self._quantum_instance = quantum_instance
+        self._circuit_sampler = CircuitSampler(
+            quantum_instance, param_qobj=is_aer_provider(quantum_instance.backend)
+        )
+
+    def _evolve_helper(
+        self,
+        init_state_param_dict: Dict[Parameter, complex],
+        hamiltonian: OperatorBase,
+        time: float,
+        t_param: Parameter,
+        initial_state: Optional[Union[OperatorBase, QuantumCircuit]] = None,
+        error_calculator=None,
+    ) -> OperatorBase:
+        r"""
+        Helper method for performing time evolution. Works both for imaginary and real case.
+
+        Args:
+            init_state_param_dict: Parameter dictionary with initial values for a given
+                parametrized state/ansatz.
+            hamiltonian:
+                Operator used for Variational Quantum Imaginary Time Evolution (VarQTE)
+                The coefficient of the operator (``operator.coeff``) determines the evolution
+                time.
+                The operator may be given either as a composed op consisting of a Hermitian
+                observable and a ``CircuitStateFn`` or a ``ListOp`` of a ``CircuitStateFn`` with a
+                ``ComboFn``.
+                The latter case enables the evaluation of a Quantum Natural Gradient.
+            time: Total time of evolution.
+            t_param: Time parameter in case of a time-dependent Hamiltonian.
+            initial_state: Quantum state to be evolved.
+            error_calculator: Calculator of errors for error-based ODE functions.
```

Comment:
Maybe it would be good to mention this is not yet supported
```suggestion
            error_calculator: Not yet supported. Calculator of errors for error-based ODE functions.
```